src/sentry/options/types.py

```diff
@@ -0,0 +1,173 @@
+"""
+sentry.options.types
+~~~~~~~~~~~~~~~~~~~~
+
+:copyright: (c) 2010-2015 by the Sentry Team, see AUTHORS for more details.
+:license: BSD, see LICENSE for more details.
+"""
+from __future__ import absolute_import, print_function
+
+from sentry.utils.yaml import safe_load
+from yaml.parser import ParserError
+from yaml.scanner import ScannerError
+
+__all__ = (
+    'InvalidTypeError',
+    'Any', 'Bool', 'Int', 'Float', 'String', 'Dict', 'Sequence',
+)
+
+
+class InvalidTypeError(TypeError):
+    pass
+
+
+class OptionType(object):
+    """Base OptionType that provides type coersion"""
+    name = ''
+    # Default value to be returned when initializing
+    default = None
+    # Types that do not need to be coerced
+    expected_types = (object,)
+    # Types that are acceptable for coersion
+    compatible_types = (basestring,)
+
+    def __call__(self, value=None):
+        if value is None:
+            return self._default()
+        if self.test(value):
+            return value
+        if isinstance(value, self.compatible_types):
+            rv = self.convert(value)
+            # Make sure convert was able to do the right thing
+            # and give us the type we were expecting
+            if self.test(rv):
+                return rv
```

Comment:
The error raised here on fallthrough is confusing because the conversion function returned an invalid value, not the user. Maybe this should be an assertion? (More notes on this below.)

In my opinion, if convert can't convert the value, it should be raising an exception rather than returning without issue.